packages/ui/src/components/DatePicker/PreviousDateRangePicker.tsx

```diff
@@ -60,6 +66,74 @@ function toDateTimeString(date: Date, showYear: boolean) {
 	return date.toLocaleDateString('en-us', options)
 }
 
+/**
+ * Takes in a time string in the format HH:mm aa or HH:mmaa and returns	an object with the hour, minute, time of day, and 24 hour time
+ * @param timeString
+ * @returns {	hour: number, minute: number, timeOfDay: string, hour24: number}
+ */
+export const getTimeInfo = (timeString: string) => {
+	const time = timeString.slice(0, -2).trim()
+	const period = timeString.slice(-2).trim().toUpperCase()
+	const [hour, minute] = time.split(':')
+	const isAM = period === 'AM'
+	const isPM = period === 'PM'
+	const hourInt = parseInt(hour, 10)
+	const hour24 =
+		isPM && hourInt !== 12
+			? hourInt + 12
+			: isAM && hourInt === 12
+			? 0
+			: hourInt
+	const timeOfDay = isAM ? 'AM' : isPM ? 'PM' : ''
+
+	return { hour: hourInt, minute: parseInt(minute, 10), timeOfDay, hour24 }
+}
+
+/**
+ * Takes in a date and a time string in the format HH:mm aa or HH:mmaa and returns a new date with the time set
+ * @param date Date
+ * @param hour number
+ * @param minute number
+ * @returns new date	with time set
+ */
+const setTimeOnDate = (date: Date, hour: number, minute: number) => {
+	const newDate = new Date(date)
+
+	newDate.setHours(hour)
+	newDate.setMinutes(minute)
+
+	return newDate
+}
+
+/**
+ * Converts a Date object to a string in the format "HH:mm aa".
+ *
+ * @param {Date} date - The Date object to convert.
+ * @returns {string} A string in the format "HH:mm aa".
+ */
+const getTimeStringFromDate = (date: Date): string => {
+	if (!date) return '00:00 AM'
+
+	const hour = date.getHours() % 12 || 12
+	const minute = date.getMinutes().toString().padStart(2, '0')
+	const period = date.getHours() < 12 ? 'AM' : 'PM'
+
+	return `${hour}:${minute} ${period}`
+}
```

Comment:
`moment().format()` would be a cleaner way to do that.
the reason we prefer to use a library for something like this is it is something well tested that covers all the edge-cases well. this would be a new implementation of something that already exists that we would need to test and maintain.

Reply:
Changed this to use moment for formatting.